src/app/qgsmaptooltrimextendfeature.cpp

```diff
@@ -221,6 +221,7 @@ void QgsMapToolTrimExtendFeature::canvasReleaseEvent( QgsMapMouseEvent *e )
         {
           getPoints( match, pLimit1, pLimit2 );
           mStep = StepExtend;
+          mLimitLayer = match.layer();
```

Comment:
Even if I understand there is no change to happen, maybe it will be better to reset mLimitLayer to 0 on deactivate

Reply:
You're right, normally there is no chance to happen, but it's good for me to reset these layers. I added other constraints too.